frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -329,11 +334,11 @@ const ExistingKMSDropDown: React.FC<ExistingKMSDropDownProps> = ({
   };
 
   const kmsProviderDropdownItems = _.reduce(
-    UISupportedProviders,
+    SupportedProviders,
     (res, providerDetails, providerName) => {
       if (
-        !UISupportedProviders[providerName].allowedPlatforms ||
-        UISupportedProviders[providerName]?.allowedPlatforms.includes(infraType)
+        !SupportedProviders[providerName].allowedPlatforms ||
+        SupportedProviders[providerName]?.allowedPlatforms.includes(infraType)
```

Comment:
```suggestion
       SupportedProviders[providerName]?.allowedPlatforms?.includes(infraType)
```
This should work as well.

Reply:
the logic is,
   By default, KMS services are supported on all infraType. If it is supported to only on limited infraType then its needs to be mentioned in allowedPlatforms list.
   
   So the conditions are, Allows even no allowedPlatforms are mentioned (or) if it mentioned then it should match with the current platform.
   
 ```SupportedProviders[providerName]?.allowedPlatforms?.includes(infraType)``` this wont work for 1st condition